### PR TITLE
Save camera coordinates and map scale

### LINKED ISSUE #304: Remember the Camera Position in UserMetaData
It would be nice when we load a map to bring you back to where you left off on the map. 

We would probably want to do this separately per map, and load on map open

### PR DESCRIPTION
**Pull request by tlein**
Resolves #304 
**Commit message:**
Saves the camera position and scale in the user meta data

`https://github.com/ild-games/duckling/issues/304`

### CODE REVIEW

**Comment by tlein** [comment]
Doesn't set the scroll position when coming back from another map

File: src/duckling/canvas/_toolbars/top-toolbar.component.ts [hunk 1]
```diff
@@ -73,6 +73,7 @@ export class TopToolbarComponent {
     @Input() mapName : string;
 
     @Output() mapSelected = new EventEmitter<String>();
+    @Output() saveClicked = new EventEmitter<void>();
 
     constructor(public store : StoreService,
                 public project : ProjectService,
```
Comment (jimmcnulty41) [inline src/duckling/canvas/_toolbars/top-toolbar.component.ts]: Looks like we don't use project anymore

File: src/duckling/canvas/canvas.component.ts [hunk 2]
```diff
@@ -139,19 +144,23 @@ export class CanvasComponent implements OnChanges, OnDestroy, AfterViewInit {
...
             this.onScaleChanged(changes.scale.previousValue, changes.scale.currentValue);
+        } 
+        if (changes.initialScrollPosition) {
+            if (this.scrollPosition != changes.initialScrollPosition.currentValue) {
+                this.scrollTo(changes.initialScrollPosition.currentValue);
+            }
         }
 
-        this._render();
+        if (this._viewInited) {
```
Comment (jimmcnulty41) [inline src/duckling/canvas/canvas.component.ts]: viewInited? That's just like difficult to pronounce

File: src/duckling/canvas/_toolbars/top-toolbar.component.ts (hunk 1 above)
Comment (tlein) [reply src/duckling/canvas/_toolbars/top-toolbar.component.ts]: fixed

**Review by jimmcnulty41 — APPROVED** [review]

**Pull request merged**